src/main/java/address/controller/RootLayoutController.java

```diff
@@ -71,9 +127,18 @@ private void handleNew() {
     @FXML
     private void handleOpen() {
         // Show open file dialog
-        File file = getXmlFileChooser().showOpenDialog(mainController.getPrimaryStage());
-        if (file == null) return;
-        EventManager.getInstance().post(new LoadDataRequestEvent(file));
+        File toOpen = getXmlFileChooser().showOpenDialog(mainController.getPrimaryStage());
+        if (toOpen == null) return;
+        PrefsManager.getInstance().setSaveLocation(toOpen);
+        EventManager.getInstance().post(new LoadDataRequestEvent(toOpen));
+    }
+
+    @FXML
+    private void handleChooseMirror() {
+        // Show open file dialog
+        File toSyncWith = getXmlFileChooser().showOpenDialog(mainController.getPrimaryStage());
+        if (toSyncWith == null) return;
```

Comment:
The `FileChooser.showOpenDialog` API returns `null` if the file choosing action was cancelled, we will have to catch the `null` as close to the call as possible.

I will change the `getXmlFileChooser` method to `getFileChosenByUser` or something like that and catch the null there instead.